api4/apitestlib.go

```diff
@@ -147,18 +145,48 @@ func setupTestHelper(enterprise bool, updateConfig func(*model.Config)) *TestHel
 	return th
 }
 
-func SetupEnterprise() *TestHelper {
+func SetupEnterprise(tb testing.TB) *TestHelper {
```

Comment:
we are already using that approach, the thing is, if the user is setting up the environment with `Setup` instead of `UnitSetup` i going to skip the test if you are running with `-short`. Basically what I'm doing here is, if you are going to need to instantiate the real database and the flag `-short` is passed, skip that test. That makes a lot of tests to run faster because doesn't need to clean up the database.

The logic for `UnitSetup` and `Setup` is outside the `testing.Short()`, because is exactly what give as the posibility of distinguish between tests that use database or not, and then allow to skip them automatically.